File: java/oop/ContactManagerApp/src/com/techlabs/contactmanager/ContactManager.java
package com.techlabs.contactmanager;

import java.io.FileNotFoundException;
import java.io.IOException;
import java.util.ArrayList;

public class ContactManager{

	private ArrayList<Contact> contacts;
	
	public ContactManager() throws FileNotFoundException, ClassNotFoundException, IOException{
		this.contacts=StoreContact.getObjectFromFile();
	}
	
	public void addContact(String name,String number) throws FileNotFoundException, IOException{
		this.contacts.add(new Contact(name,number));
		StoreContact.saveObjectToFile(this.contacts);
	}
	
	public ArrayList<Contact> getContacts(){
		return this.contacts;
	}
	
}

File: dotNet/CSharp/ContactManagerApp/ContactManagerApp/ContactManager.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Runtime.Serialization;
using System.Runtime.Serialization.Formatters.Binary;
using System.Text;

namespace ContactManagerApp
{
    class ContactManager
    {
        private List<Contact> _contacts;

        public ContactManager()
        {
            _contacts = StoreContact.getObjectFromFile();
        }

        public void AddContact(string name,string number)
        {
            _contacts.Add(new Contact(name, number));
            StoreContact.saveObjectToFile(_contacts);
        }

        public List<Contact> Contacts
        {
            get
            {
                return _contacts;
            }
        }
    }
}
